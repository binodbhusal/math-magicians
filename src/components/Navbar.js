import React from 'react';
import { Link } from 'react-router-dom';
import styles from './Navbar.module.css';

const Navbar = () => (
  <>
    <h1>Math Magicians</h1>

    <nav>
      <ul className={styles.navul}>
        <li className={styles.navlink}>
          <Link to="/" className={styles.navlink}>Home</Link>
        </li>
        <li className={styles.navlink}>
          <Link to="/calculator" className={styles.navlink}>Calculator</Link>
        </li>
        <li className={styles.navlink}>
          <Link to="/quote" className={styles.navlink}>Qoute</Link>
        </li>
      </ul>
    </nav>
  </>
);
export default Navbar;
