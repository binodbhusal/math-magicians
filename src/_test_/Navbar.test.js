import React from 'react';
import { render, screen } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import Navbar from '../components/Navbar';
import '@testing-library/jest-dom/extend-expect';

test('Navbar renders correctly', () => {
  render(
    <BrowserRouter>
      <Navbar />
    </BrowserRouter>,
  );

  const homeLink = screen.getByText('Home');
  const calculatorLink = screen.getByText('Calculator');
  const quoteLink = screen.getByText('Qoute');

  expect(homeLink).toBeInTheDocument();
  expect(calculatorLink).toBeInTheDocument();
  expect(quoteLink).toBeInTheDocument();
});
