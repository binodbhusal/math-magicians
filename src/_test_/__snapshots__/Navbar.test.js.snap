// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Expect Navbar to match working snapshot 1`] = `
<DocumentFragment>
  <h1>
    Math Magicians
  </h1>
  <nav>
    <ul
      class="navul"
    >
      <li
        class="navlink"
      >
        <a
          class="navlink"
          href="/"
        >
          Home
        </a>
      </li>
      <li
        class="navlink"
      >
        <a
          class="navlink"
          href="/calculator"
        >
          Calculator
        </a>
      </li>
      <li
        class="navlink"
      >
        <a
          class="navlink"
          href="/quote"
        >
          Qoute
        </a>
      </li>
    </ul>
  </nav>
</DocumentFragment>
`;
